src/utils/components/SSHAccess/components/SSHServiceSelect.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC } from 'react';
+
+import { IoK8sApiCoreV1Service } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { useK8sModels } from '@openshift-console/dynamic-plugin-sdk';
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import { METALLB_GROUP, SERVICE_TYPES } from '../constants';
+
+type SSHServiceSelectProps = {
+  sshService: IoK8sApiCoreV1Service;
+  sshServiceLoaded: boolean;
+  onSSHChange: (serviceType: SERVICE_TYPES) => void;
+};
+
+const SSHServiceSelect: FC<SSHServiceSelectProps> = ({
+  sshService,
+  sshServiceLoaded,
+  onSSHChange,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [models] = useK8sModels();
+
+  const hasSomeMetalCrd = Object.keys(models).some((modelGroupVersionKind) =>
+    modelGroupVersionKind.startsWith(METALLB_GROUP),
+  );
+
+  const handleChange = (event: React.ChangeEvent<Element>, newValue: string | undefined) => {
+    setIsOpen(false);
+
+    if (newValue === sshService?.spec?.type) return;
+    onSSHChange(newValue as SERVICE_TYPES);
+  };
+
+  return (
+    <Select
+      menuAppendTo="parent"
+      isOpen={isOpen}
+      onToggle={setIsOpen}
+      onSelect={handleChange}
+      variant={SelectVariant.single}
+      selections={sshService?.spec?.type ?? SERVICE_TYPES.NONE}
+      isDisabled={!sshServiceLoaded}
+    >
+      <SelectOption value={SERVICE_TYPES.NONE}>{t('None')}</SelectOption>
+      <SelectOption
+        value={SERVICE_TYPES.NODE_PORT}
+        description={t(
+          'Opens a specific port on all Nodes in the cluster. If the Node is publicly accessible, any trtaffic that is sent to this port is forwarded to the Service',
```

Comment:
the logic to prevent onSSHChange from firing unnecessarily is [here](https://github.com/kubevirt-ui/kubevirt-plugin/pull/1036/files#diff-4733b7d6e33fb5a1cfc1fb47ee70a01dede35dbf7794de437c6948fddfb69703R38). There was a bug. Fixed I think :D